zombie-monitor/src/main/scala/com/broadinstitute/dsp/zombieMonitor/DeletedOrErroredNodepoolChecker.scala

```diff
@@ -9,11 +9,13 @@ import io.chrisdavenport.log4cats.Logger
 import org.broadinstitute.dsde.workbench.model.TraceId
 
 /**
- * Scans through all active nodepools and check if they still exist in Google.
- * If they're deleted from Google, we're updating them to `DELETED` in leonardo DB.
- * We're also updating any kubernetes apps running on these nodepools as `DELETED`.
+ * Scans through all active nodepools, and update DB accordingly when necessary
+ *
+ * - if nodepool doesn't exist in google anymore, mark it as `DELETED`, mark associated APP as `DELETED` and report this nodepool
+ * - If nodepool exist in google in ERROR, mark it as `ERROR`, mark associated APP as `ERROR`  and report this nodepool
```

Comment:
I wonder if we should report the app as well 🤔

Reply:
it'll be a bit more complicated if we want to report `App` as well since for now, for a given report file, we assume they're all the same type (like runtimes, disks, nodepools). so it'll require a bit more thinking how we organize things. so I think we should punt this for now.

Reply:
Punting for now makes sense. Should we create a ticket so we don't forget?

Reply:
on a second thought, I'm not sure if it's that useful to report `App`, if we have a specific app that we're wondering if it being updated properly, we can always go to its nodepool, and check the report to see if it was indeed updated by `janitor` (currently resource-validator).. all reports contain the source object, which is pretty easy to understand. If in the future, we have a use case where we do care about `App`, then I think it makes more sense to have a ticket...

Reply:
I thought it could debugging a bit faster in the case of UL questions, etc. but yea, it doesn't seem like a big deal right now since we have 1-to-1 mapping between apps and nodepools.